api/api/settings.py

```diff
@@ -97,6 +97,21 @@
     'EXCEPTION_HANDLER': 'desecapi.exception_handlers.exception_handler',
     'DEFAULT_VERSIONING_CLASS': 'rest_framework.versioning.NamespaceVersioning',
     'ALLOWED_VERSIONS': ['v1', 'v2'],
+    'DEFAULT_THROTTLE_CLASSES': [
+        'desecapi.throttling.ScopedRatesThrottle',
+        'rest_framework.throttling.UserRateThrottle',
+    ],
+    'DEFAULT_THROTTLE_RATES': {
+        # ScopedRatesThrottle
+        'account_management_active': ['3/min'],  # things that send email or create domains on pdns
+        'account_management_passive': ['10/min'],  # things like viewing your account or creating/deleting tokens
+        'dyndns': ['1/min'],  # dynDNS updates; anything above 1/min is a client misconfiguration
+        'dns_api_read': ['5/s', '50/min'],  # DNS API requests that do not involve pdns
+        'dns_api_write': ['3/s', '50/min', '200/h'],  # DNS API requests that do involve pdns
+        # UserRateThrottle
+        'user': '1000/d',  # hard limit on requests by a) an authenticated user, b) an unauthenticated IP address
+    },
+    'NUM_PROXIES': 0,  # Do not use X-Forwarded-For header when determining IP for throttling
```

Comment:
then all requests will be considered coming from nginx's IP address?

Reply:
Doesn't the API see the client IP directly? We use it [here](https://github.com/desec-io/desec-stack/blob/4136abc/api/desecapi/views.py#L294).

I wonder why that is. Does [`uwsgi_pass`](https://github.com/desec-io/desec-stack/blob/c07c4ea/www/conf/sites-available/90-desec.api.location#L13) do the magic?

Reply:
Indeed, it's a combination of our [uwsgi configuration](https://uwsgi-docs.readthedocs.io/en/latest/Nginx.html#what-is-the-uwsgi-params-file) and nginx's [remote_addr variable](http://nginx.org/en/docs/http/ngx_http_core_module.html#var_remote_addr)